### PR TITLE
feat: added additional ChainId prop for mock

### PR DESCRIPTION
**Pull request by chrismaree**
- changes for npm release (#70)
- update events after internal chat (#67)
- feat: Add gas analysis scripts for SpokePool methods (#61)
- feat: Make token bridge programmable per tokens (#65)
- improve(spoke-pool): Slow relay execution should emit FillRelay event (#69)
- fix(optimism-adapter): handle custom bridges (#68)
- improve: Add ability to unwhitelist a route (#66)
- improve: Add isSlowRelay param to FilledRelay event (#75)
- fix: Check if spoke pool is initialized before executing root bundle (#76)
- improve: Update ethereum package versions (#72)
- feat: make slow relay refund the user realyerFeePct (#77)
- improve(buildSampleTree): Make script more customizable (#71)
- build for release, including exporting artefacts and fixtures (#74)
- feat: add additional helpers used in relayer tests (#80)
- feat: add emergency admin functions (#78)
- feat: added additional ChainId prop for mock

### CODE REVIEW

File: contracts/test/MockSpokePool.sol [hunk 1]
```diff
@@ -9,14 +9,24 @@ import "../SpokePoolInterface.sol";
...
+    uint256 chainId_;
+
     constructor(
         address _crossDomainAdmin,
         address _hubPool,
         address _wethAddress,
-        address timerAddress
-    ) SpokePool(_crossDomainAdmin, _hubPool, _wethAddress, timerAddress) {}
+        address timerAddress,
+        uint256 _chainId
```
Comment (nicholaspai) [inline contracts/test/MockSpokePool.sol:19]: As discussed IRL would be nice not to have to pass in a new constructor param. Instead default this to 0 and allow someone to override it

**Review by nicholaspai — APPROVED** [review]

File: test/SpokePool.Deposit.ts [hunk 2]
```diff
@@ -117,7 +117,7 @@ describe("SpokePool Depositor Logic", async function () {
     const currentSpokePoolTime = await spokePool.getCurrentTime();
 
     // Blocked if user hasn't approved token.
-    await erc20.connect(depositor).approve(spokePool.address, 0);
+    await erc20.connect(depositor).approve(spokePool.address);
```
Comment (nicholaspai) [inline test/SpokePool.Deposit.ts:120]: why change this?

Comment (chrismaree) [reply test/SpokePool.Deposit.ts:120]: mistake. have reverted

File: contracts/test/MockSpokePool.sol [hunk 3]
```diff
@@ -19,4 +21,13 @@ contract MockSpokePool is SpokePoolInterface, SpokePool {
     function _bridgeTokensToHubPool(RelayerRefundLeaf memory relayerRefundLeaf) internal override {}
 
     function _requireAdminSender() internal override {}
+
+    function chainId() public view override(SpokePool, SpokePoolInterface) returns (uint256) {
```
Comment (nicholaspai) [inline contracts/test/MockSpokePool.sol:25]: nit you can remove the override of the `SpokePoolInterface` if you remove this from the contract. I don't think the other SpokePools extend `SpokePoolInterface` anyways

**Pull request merged**